tests/gatewayapitests/status/httproute_test.go

```diff
@@ -851,3 +851,177 @@ func TestHTTPRouteWithOneExistingAndOneNonExistingGateway(t *testing.T) {
+	condition = metav1.Condition{
+		Type:    string(gatewayv1.RouteConditionAccepted),
+		Reason:  string(gatewayv1.RouteReasonAccepted),
+		Status:  metav1.ConditionTrue,
+		Message: "Parent reference is valid",
+	}
+
+	conditions = append(conditions, condition)
+	conditionMap[fmt.Sprintf("%s-%d", gatewayName, ports[1])] = conditions
+
+	conditionMap["gateway-hr-14-8081"][0].Message = "Matching gateway listener is in Invalid state"
+	conditionMap["gateway-hr-14-8081"][0].Status = metav1.ConditionFalse
+	conditionMap["gateway-hr-14-8081"][0].Reason = string(gatewayv1.RouteReasonPending)
+
+	expectedRouteStatus = akogatewayapitests.GetRouteStatusV1([]string{gatewayName}, namespace, []int32{ports[1]}, conditionMap)
+	akogatewayapitests.ValidateHTTPRouteStatus(t, &httpRoute.Status, &gatewayv1.HTTPRouteStatus{RouteStatus: *expectedRouteStatus})
+
+	akogatewayapitests.TeardownHTTPRoute(t, httpRouteName2, namespace)
+	akogatewayapitests.TeardownGateway(t, gatewayName, namespace)
+	akogatewayapitests.TeardownGatewayClass(t, gatewayClassName)
+}
+
+func TestHttpRouteWithValidAndInvalidGatewayListeners(t *testing.T) {
+	gatewayClassName := "gateway-class-hr-15"
+	gatewayName := "gateway-hr-15"
+	httpRouteName := "httproute-15"
+	namespace := "default"
+	ports := []int32{8080, 8081}
+
+	akogatewayapitests.SetupGatewayClass(t, gatewayClassName, akogatewayapilib.GatewayController)
+
+	listeners := akogatewayapitests.GetListenersV1(ports, false, false)
+	listeners[1].Protocol = "TCP"
+
+	g := gomega.NewGomegaWithT(t)
+	akogatewayapitests.SetupGateway(t, gatewayName, namespace, gatewayClassName, nil, listeners)
+	g.Eventually(func() bool {
+		gateway, err := akogatewayapitests.GatewayClient.GatewayV1().Gateways(namespace).Get(context.TODO(), gatewayName, metav1.GetOptions{})
+		if err != nil || gateway == nil {
+			t.Logf("Couldn't get the gateway, err: %+v", err)
+			return false
+		}
+		return apimeta.FindStatusCondition(gateway.Status.Conditions, string(gatewayv1.GatewayConditionAccepted)) != nil
+	}, 30*time.Second).Should(gomega.Equal(true))
+
+	parentRefs := akogatewayapitests.GetParentReferencesV1([]string{gatewayName}, namespace, ports)
+	hostnames := []gatewayv1.Hostname{"foo-8080.com", "foo-8081.com"}
+
+	akogatewayapitests.SetupHTTPRoute(t, httpRouteName, namespace, parentRefs, hostnames, nil)
+	g.Eventually(func() bool {
+		httpRoute, err := akogatewayapitests.GatewayClient.GatewayV1().HTTPRoutes(namespace).Get(context.TODO(), httpRouteName, metav1.GetOptions{})
+		if err != nil || httpRoute == nil {
+			t.Logf("Couldn't get the HTTPRoute, err: %+v", err)
+			return false
+		}
+		if len(httpRoute.Status.Parents) != 2 {
+			return false
+		}
+		return apimeta.FindStatusCondition(httpRoute.Status.Parents[0].Conditions, string(gatewayv1.RouteConditionAccepted)) != nil
+	}, 30*time.Second).Should(gomega.Equal(true))
+
+	conditionMap := make(map[string][]metav1.Condition)
+
+	for _, port := range ports {
+		conditions := make([]metav1.Condition, 0, 1)
```

Comment:
instead of make and append this can be refactored to 
conditions := []metav1.Condition{ {Type: ..., Reason...} }
conditionMap[fmt.Sprintf("%s-%d", gatewayName, port)] = conditions

Reply:
Done